slug.go

```diff
@@ -297,6 +279,58 @@ func (p *Packer) packWalkFn(root, src, dst string, tarW *tar.Writer, meta *Meta,
 	}
 }
 
+func (p *Packer) resolveLink(root string, path string, header *tar.Header) (*resolvedSymlink, error) {
+	// Read the symlink file to find the destination.
+	target, err := os.Readlink(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read symlink %q: %w", path, err)
+	}
+
+	// Ensure the target is acceptable per the Packer's configuration.
+	if err := p.checkSymlink(root, path, target); err != nil {
+		// Check if dereferencing is enabled. If so, we're going to
+		// try copying the symlink's data. If not, this is an error.
+		if !p.dereference {
+			return nil, err
+		}
+	} else {
+		if header != nil {
```

Comment:
Ah this code snuck in because I wanted to re-use this logic in #36 where there wouldn't be a header available -- so no, the header should never be `nil` since it is initialized in L208.

However, I've decided against allowing for the root path to be a chain of links. In the context of Terraform, someone should trigger an apply from the link that resolves to the actual working directory.